cmake/app/boilerplate.cmake

```diff
@@ -282,6 +282,9 @@ if(DEFINED BOARD_REVISION)
   string(REPLACE "." "_" BOARD_REVISION_STRING ${BOARD_REVISION})
 endif()
 
+# The BOARD variable is now set to its final value.
+zephyr_boilerplate_watch(BOARD)
```

Comment:
Let's remove this check here, and instead monitor values where we also checks the cache.

That is, move this check inside:
https://github.com/zephyrproject-rtos/zephyr/blob/d8283b63ec6fe9aa1948db163923afc0594a2bde/cmake/extensions.cmake#L2047

so that any variable that must not be changed in the cache are also watched automatically.

That is, right above this line:
https://github.com/zephyrproject-rtos/zephyr/blob/d8283b63ec6fe9aa1948db163923afc0594a2bde/cmake/extensions.cmake#L2111
```
   zephyr_boilerplate_watch(${variable}) 
endfunction(zephyr_check_cache variable) 
```

Reply:
> That is, move this check inside:

Done for both BOARD and SHIELD.